storage/aptosdb/src/state_store/mod.rs

```diff
@@ -611,17 +612,24 @@ impl StateStore {
             .with_label_values(&["add_state_kv_batch"])
             .start_timer();
 
-        value_state_sets
+        sharded_state_kv_batches
             .par_iter()
             .enumerate()
-            .flat_map_iter(|(i, kvs)| {
-                let version = first_version + i as Version;
-                kvs.iter().map(move |(k, v)| {
-                    sharded_state_kv_batches[k.get_shard_id() as usize]
-                        .put::<StateValueSchema>(&(k.clone(), version), v)
-                })
-            })
-            .collect()
+            .try_for_each(|(shard_id, batch)| {
+                value_state_sets
+                    .par_iter()
+                    .enumerate()
+                    .flat_map_iter(|(i, shards)| {
+                        let version = first_version + i as Version;
+                        let kvs = &shards[shard_id];
+                        kvs.iter().map(move |(k, v)| {
```

Comment:
why is `move` needed?

Reply:
move the `version`, otherwise it doesn't compile.